server/secrets.go

```diff
@@ -0,0 +1,98 @@
+package server
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/docker/docker/pkg/archive"
+	rspec "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/opencontainers/selinux/go-selinux/label"
+	"github.com/pkg/errors"
+	pb "k8s.io/kubernetes/pkg/kubelet/apis/cri/v1alpha1/runtime"
+)
+
+const (
+	// defaultmountCfgFile holds the default mount paths in the form "host:container"
+	defaultMountCfgFile = "/usr/share/containers/mounts.conf"
+	// overridetmountCfgFile holds the override mount paths in the form "host:container"
+	overrideMountCfgFile = "/etc/containers/mounts.conf"
+)
+
+// readMountFile returns a list of the host:container paths
+func readMountFile(mountFilePath string) ([]string, error) {
+	var mountPaths []string
+	file, err := os.Open(mountFilePath)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	scanner.Split(bufio.ScanLines)
+	for scanner.Scan() {
+		mountPaths = append(mountPaths, scanner.Text())
+	}
+
+	return mountPaths, nil
+}
+
+// getHostAndCtrDir separates the host:container paths
+func getHostAndCtrDir(path string) (string, string, error) {
+	arr := strings.SplitN(path, ":", 2)
+	if len(arr) == 2 {
+		return arr[0], arr[1], nil
+	}
+	return "", "", errors.Errorf("unable to get host and container dir")
+}
+
+// secretMount copies the contents of host directory to container directory
+// and returns a list of mounts
+func secretMounts(mountLabel, mountFilePath string) ([]rspec.Mount, error) {
+	var mounts []rspec.Mount
+	mountPaths, err := readMountFile(mountFilePath)
+	if err != nil {
+		return nil, err
+	}
+	for _, path := range mountPaths {
+		hostDir, ctrDir, err := getHostAndCtrDir(path)
+		if err != nil {
+			return nil, err
+		}
+
+		if err := os.RemoveAll(ctrDir); err != nil {
+			return nil, fmt.Errorf("remove container directory failed: %v", err)
+		}
+
+		if err := os.MkdirAll(ctrDir, 0755); err != nil {
+			return nil, fmt.Errorf("making container directory failed: %v", err)
+		}
+
+		stuff, err := archive.Tar(hostDir, archive.Uncompressed)
```

Comment:
??? why tarring this up?

Reply:
It is easier to copy the content from host to container. tar up the contents in the host dir and untar it into the container dir.

Reply:
I thought we were tarring up the contents in docker?

Reply:
no, it is recursive copy in docker. But I liked the idea of tarring up better so I went with that. Changing it back to the way it is done in docker

Reply:
Ok

Reply:
fixed.